tests/Unit/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/Test_ChangeSlabSize.cpp

```diff
@@ -0,0 +1,213 @@
+  // the coordinates are never actually used, just the size of the map. Here we
+  // use 2 elements.
+  std::unordered_map<ElementId<Dim>, tnsr::I<DataVector, Dim, Frame::Grid>>
+      element_faces_grid_coords{};
+  element_faces_grid_coords[element_ids.at(0)] =
+      tnsr::I<DataVector, Dim, Frame::Grid>{};
+  element_faces_grid_coords[element_ids.at(1)] =
+      tnsr::I<DataVector, Dim, Frame::Grid>{};
+
+  // we need to create a new runner every time an ASSERT is triggered because
+  // they can not be copied.
+  const auto create_new_runner =
+      [&time_step_id_1, &time_stepper, &step_1, &start_time_1,
+       &element_faces_grid_coords](
+          gsl::not_null<
+              ActionTesting::MockRuntimeSystem<MockMetavariables<Dim>>*>
+              runner) {
+        ActionTesting::emplace_component_and_initialize<worldtube_chare>(
+            runner, 0,
+            {time_step_id_1, time_stepper->next_time_id(time_step_id_1, step_1),
+             step_1, step_1, start_time_1.value(),
+             std::make_unique<TimeSteppers::Rk3HesthavenSsp>(),
+             element_faces_grid_coords});
+        ActionTesting::set_phase(runner, Parallel::Phase::Testing);
+      };
+  const auto check_time_tags =
+      [](ActionTesting::MockRuntimeSystem<MockMetavariables<Dim>>& runner,
+         const TimeStepId& time_step_id, const TimeStepId& next_time_step_id,
+         const TimeDelta& step, const TimeDelta& next_step, const double time) {
+        CHECK(
+            ActionTesting::get_databox_tag<worldtube_chare, ::Tags::TimeStepId>(
+                runner, 0) == time_step_id);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare,
+                                             ::Tags::Next<::Tags::TimeStepId>>(
+                  runner, 0) == next_time_step_id);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare, ::Tags::TimeStep>(
+                  runner, 0) == step);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare,
+                                             ::Tags::Next<::Tags::TimeStep>>(
+                  runner, 0) == next_step);
+        CHECK(ActionTesting::get_databox_tag<worldtube_chare, ::Tags::Time>(
+                  runner, 0) == time);
+      };
+  using inbox_variables_type =
+      Variables<tmpl::list<CurvedScalarWave::Tags::Psi,
+                           ::Tags::dt<CurvedScalarWave::Tags::Psi>>>;
+  {
+    ActionTesting::MockRuntimeSystem<MockMetavariables<Dim>> runner{{}};
+    create_new_runner(make_not_null(&runner));
+    auto& worldtube_inbox =
+        ActionTesting::get_inbox_tag<worldtube_chare,
+                                     Tags::SphericalHarmonicsInbox<Dim>>(
+            make_not_null(&runner), 0);
+    worldtube_inbox[time_step_id_1][element_ids.at(0)] = inbox_variables_type{};
+    // ChangeSlabSize but only 1/2 elements have sent
+    CHECK_FALSE(ActionTesting::next_action_if_ready<worldtube_chare>(
+        make_not_null(&runner), 0));
```

Comment:
Check the result of calling the action here.  I think this is when the change is expected (or in this case not expected since there's nothing to do).